docs/user-guides/client-api-config.md

```diff
@@ -0,0 +1,676 @@
+# Client API Config
+
+Vald provides client libraries for indexing vectors, searching approximate nearest neighbor vectors, and removing indexing vectors.
+Each request allows setting request configuration for user demands.
+Please select the section you need.
+
+Please also see each API document.
+
+## Insert Service
+
+`Insert` is inserting new vectors into the Vald cluster.
+It requires the vector, its ID (specific ID for the vector), and optional configuration.
+
+### Configuration
+
+```rpc
+// Represent insert configuration.
+message Config {
+  // Check whether or not the same set of vector and ID is already inserted.
+  bool skip_strict_exist_check = 1;
+  // Configuration for filters if your Vald cluster uses filters.
+  Filter.Config filters = 2;
+  // The timestamp when the vector was inserted.
+  int64 timestamp = 3;
+}
+```
+
+<details><summary>Insert Configuration Sample (Go)</summary><br>
+
+```go
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald-client-go/v1/payload"
+	"github.com/vdaas/vald-client-go/v1/vald"
+	"google.golang.org/grpc"
+)
+
+func main() {
+	// Create connection
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	target := "localhost:8080"
+	conn, err := grpc.DialContext(ctx, target)
+	if err != nil {
+		panic(err)
+	}
+	// Init Vald client
+	client := vald.NewValdClient(conn)
+
+	// Insert sample
+	location, err := client.Insert(ctx, &payload.Insert_Request{
+		// Vector information (mandatory)
+		Vector: &payload.Object_Vector{
+			// Set the vector
+			Vector: []float32{0, 1, 2},
+			// Set the specific ID for the vector
+			// The ID must not be indexed.
+			Id:     "sample",
+		},
+		// Insert configuration (optional)
+		Config: &payload.Insert_Config{
+			SkipStrictExistCheck: false,
+			Filters: &payload.Filter_Config{
+				Targets: []*payload.Filter_Target{
+					{
+						Host: "vald-ingress-filter",
+						Port: 8081,
+					},
+				},
+			},
+			Timestamp: time.Now().UnixMilli(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	...
+}
+```
+
+</details>
+
+#### skip_strict_exist_check
+
+`skip_strict_exist_check` is a flag to skip checking whether the same set of the vector and ID is already inserted or not.
+If it is set as `true`, the checking function will be skipped.<BR>
+The default value is `false`.
+
+#### filters
+
+`filters` is the configuration when using filter functions.
+In the `Insert` section, it is popular to use ingress filtering.
```

Comment:
Is it better to use `common` ? 🤔

```suggestion
In the `Insert` section, it is common to use ingress filtering.
```

Reply:
https://github.com/vdaas/vald/pull/1783/commits/0c006116b16c5050a6204aa32b3d4680c9a153ce